sqlstore/src/it/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateListPoolViewDaoIT.java

```diff
@@ -32,18 +34,27 @@ public void setup() throws IOException {
         sut.setEntityManager(entityManager);
     }
 
-    public void assertWorksetHasPools(long worksetId, int expectedCount) {
+    public void assertWorksetHasPools(long worksetId, int expectedCount, long... expectedPoolIds) {
         QueryBuilder<ListPoolView, ListPoolView> builder = new QueryBuilder<>(currentSession(), ListPoolView.class, ListPoolView.class);
         sut.restrictPaginationByWorksetId(builder, worksetId, msg -> fail(msg));
 
         List<ListPoolView> results = builder.getResultList();
         assertEquals(worksetId, expectedCount, results.size());
+        List<Long> actualPoolIds = new ArrayList<>();
+        for(ListPoolView row: results){
+            actualPoolIds.add(row.getId());
+        }
+
+        for(long expectedId: expectedPoolIds){
```

Comment:
Since you're now passing in the `expectedPoolIds`, you could drop the `expectedCount` arg (since it's equal to the length if `expectedPoolIds`)